lib/shared/bucketing-assembly-script/assembly/types/configBody.ts

```diff
@@ -13,30 +13,44 @@ import { Feature } from './feature'
 import { Audience } from './target'
 
 export class PublicProject extends JSON.Value {
-    readonly _id: string
-    readonly key: string
-    readonly a0_organization: string
-    readonly settings: JSON.Obj
+    readonly _id: string;
+    readonly key: string;
+    readonly a0_organization: string;
+    disablePassthroughRollouts: bool; // Now part of PublicProject directly
 
     constructor(project: JSON.Obj) {
-        super()
-        this._id = getStringFromJSON(project, '_id')
-        this.key = getStringFromJSON(project, 'key')
-        this.a0_organization = getStringFromJSON(project, 'a0_organization')
-        this.settings = getJSONObjFromJSON(project, 'settings')
+        super();
+        this._id = getStringFromJSON(project, '_id');
+        this.key = getStringFromJSON(project, 'key');
+        this.a0_organization = getStringFromJSON(project, 'a0_organization');
+
+        // Parse settings directly within this class
+        let settings = getJSONObjFromJSON(project, 'settings');
+        if (settings.has("disablePassthroughRollouts")) {
+            const value = settings.get("disablePassthroughRollouts");
+            if (value instanceof JSON.Val) {
+                this.disablePassthroughRollouts = !!value; // Coerce the value to boolean
+            } else {
+                console.error("Expected a boolean value for 'disablePassthroughRollouts', but received: ", value);
+                this.disablePassthroughRollouts = false; // Default to false if not correctly formatted
+            }
+        } else {
+            this.disablePassthroughRollouts = false; // Default to false if not present
+        }
     }
 
     stringify(): string {
-        const json = new JSON.Obj()
-        json.set('_id', this._id)
-        json.set('key', this.key)
-        json.set('a0_organization', this.a0_organization)
-        json.set('settings', this.settings)
+        const json = new JSON.Obj();
+        json.set('_id', this._id);
+        json.set('key', this.key);
+        json.set('a0_organization', this.a0_organization);
+        json.set('disablePassthroughRollouts', this.disablePassthroughRollouts); // Store directly
```

Comment:
I think this might be your issue, previously you were parsing and writing `settings` as a `JSON.Obj`. Now you are parsing `settings` and picking out `disablePassthroughRollouts`, then setting it as a top-level boolean.

Reply:
this was just something I tried in my latest iteration. Before this I was parsing it out into a `settings` subfield. Ill revert back to that state